### PR TITLE
[TS] LPS-135628 Allow CKEditor to set an empty text value

### PR DESCRIPTION
**Pull request by georgel-pop-lr**
[TS] LPS-135628 Allow CKEditor to set an empty text value, init instancePendingData with null and check if different than null to allow set up an empty text value.

### CODE REVIEW

**Comment by georgel-pop-lr** [comment]
ci:test:relevant

**Comment by diegonvs** [comment]
Hey @georgel-pop-lr, I'm not able to reproduce this issue in the current 7.2.x(Commit: 1ae0042b11f743e467797f87ba3e0f883e57dd03 Updated on: Tue Jul 13 12:41:15 2021 -0700).

See the following video:
https://issues.liferay.com/secure/attachment/415398/Screen%20Recording%202021-07-14%20at%2016.37.31.mov

I also left [a comment in the ticket](https://issues.liferay.com/browse/LPS-135628?focusedCommentId=2484386&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-2484386)

Am I missing something?

File: modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/ckeditor.jsp [hunk 1]
```diff
@@ -625,7 +625,7 @@ name = HtmlUtil.escapeJS(name);
 		});
 
 		ckEditor.on('dataReady', (event) => {
-			if (instancePendingData) {
+			if (instancePendingData != null) {
```
Comment (julien) [inline modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/ckeditor.jsp:628]: Not sure about the overall solution, but in any case we should **always** use strict equality, i.e. `!==` and not `!=`

Comment (georgel-pop-lr) [reply modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/ckeditor.jsp:628]: Okay, that makes sense, I will do the change.

Regards

**Comment by georgel-pop-lr** [comment]
> Hey @georgel-pop-lr, I'm not able to reproduce this issue in the current 7.2.x
> 
> (Commit: 1ae0042b11f743e467797f87ba3e0f883e57dd03 Updated on: Tue Jul 13 12:41:15 2021 -0700).
> 
> See the following video:
> https://issues.liferay.com/secure/attachment/415398/Screen%20Recording%202021-07-14%20at%2016.37.31.mov
> 
> I also left [a comment in the ticket](https://issues.liferay.com/browse/LPS-135628?focusedCommentId=2484386&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-2484386)
> 
> Am I missing something?

Yes, I forgot to specify in the steps that you must add these two props in your portal-ext.properties file:
editor.wysiwyg.default=ckeditor
editor.wysiwyg.portal-impl.portlet.ddm.text_html.ftl=ckeditor
Is only afecting CKEditor.

Regards.

**Comment by georgel-pop-lr** [comment]
ci:test:relevant

**Comment by diegonvs** [comment]
Great! I reproduced in 7.2.x. This fix looks good to me!

**Review by diegonvs — APPROVED** [review]
Great! I reproduced in 7.2.x. This fix looks good to me!

**Comment by diegonvs** [comment]
Would @markocikos leave a comment before forward this?

**Comment by markocikos** [comment]
@georgel-pop-lr I cannot reproduce the issue on `master`. From your comment in JIRA, I guess you cannot either:
> Note: Not sure with the same code I'm not able to reproduce on master, but on 7.2.x branch yes.

If this is true, I don't think we should merge this PR to `master`, rather send it only to 7.2.x / 7.3.x branches. I am not sure what is the latest workflow for patching older versions.

**Comment by georgel-pop-lr** [comment]
> @georgel-pop-lr I cannot reproduce the issue on `master`. From your comment in JIRA, I guess you cannot either:
> 
> > Note: Not sure with the same code I'm not able to reproduce on master, but on 7.2.x branch yes.
> 
> If this is true, I don't think we should merge this PR to `master`, rather send it only to 7.2.x / 7.3.x branches. I am not sure what is the latest workflow for patching older versions.

Hi @markocikos,

I have thought about the same thing you say, but as this fix is a simple one, I think it is better for code to be same between branched when possible.

Let me know what you think about this.

Regards.

**Comment by markocikos** [comment]
@georgel-pop-lr The change itself does make sense to me, allowing setting of empty value. It's not a problem to add this as a sort-of sanity check. But, why is this not an issue on `master`? Was this fixed in a different place? If so, we might want to backport the original fix. It would be preferable for the same reason, to keep similar codebase.

**Comment by georgel-pop-lr** [comment]
> @georgel-pop-lr The change itself does make sense to me, allowing setting of empty value. It's not a problem to add this as a sort-of sanity check. But, why is this not an issue on `master`? Was this fixed in a different place? If so, we might want to backport the original fix. It would be preferable for the same reason, to keep similar codebase.

Hi @markocikos,

This issue is not affecting master because we use ckeditor4-react, which behaves different, and seems to set the value with different conditions:
https://github.com/ckeditor/ckeditor4-react/blob/734aee10d11b1bb1ae748f4a5a840b2c040e127e/src/ckeditor.jsx#L36
https://github.com/ckeditor/ckeditor4-react/blob/734aee10d11b1bb1ae748f4a5a840b2c040e127e/src/ckeditor.jsx#L93

In master setHtml method https://github.com/liferay/liferay-portal/blob/e593eb3e3fe706364bc9f41ee59527c5d73fba9b/modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/ckeditor.jsp#L312 is not being used at all, so is not using the same mechanism of updating the value as of 7.3.x and below.

Is it true that setHtml method is not being used any more? If so, why it wasn't removed from master branch?

I still believe that we could add this in master just for code uniformity, let me know your thoughts.

Regards.

**Comment by markocikos** [comment]
@georgel-pop-lr I see, that makes sense. OK, let's merge it! It'll be an extra sanity check, and we'll have a bit more code uniformity over branches. Since this is basically a minor improvement, not urgent to fix, I'll put it on hold until GA release.

**Comment by georgel-pop-lr** [comment]
> @georgel-pop-lr I see, that makes sense. OK, let's merge it! It'll be an extra sanity check, and we'll have a bit more code uniformity over branches. Since this is basically a minor improvement, not urgent to fix, I'll put it on hold until GA release.

Okay great, sounds good to me!

**Comment by markocikos** [comment]
ci:forward

**Pull request closed without merging**